OpenTokMeet/app/src/main/java/com/opentok/android/meet/HomeActivity.java

```diff
@@ -61,84 +75,191 @@ public void onCreate(Bundle savedInstanceState) {
-
-        SharedPreferences settings = getApplicationContext()
-                .getSharedPreferences(LAST_CONFERENCE_DATA, 0);
-        SharedPreferences.Editor editor = settings.edit();
-        editor.putString("roomName", roomName);
-        editor.putString("username", username);
-
+    private void saveSettings() {
+        SharedPreferences.Editor editor     = null;
+        /* Conference Settings */
+        editor = this.getSharedPreferences(PREF_TAG_CONFERENCE, 0).edit();
+        editor.putString(PREF_ROOMNAME, mRoomName);
+        editor.putString(PREF_USERNAME, mUsername);
+        editor.putString(PREF_CAPTURERRES, mCapturerResolution);
+        editor.putString(PREF_CAPTURERFPS, mCapturerFps);
+        editor.putBoolean(PREF_H264, mH264Support);
         editor.apply();
     }
 
-    private void restoreConferenceData() {
-        SharedPreferences settings = getApplicationContext()
-                .getSharedPreferences(LAST_CONFERENCE_DATA, 0);
-        roomName = settings.getString("roomName", "");
-        username = settings.getString("username", "");
+    private void restoreSettings() {
+        SharedPreferences settings = null;
+        /* Conference Settings */
+        settings = this.getSharedPreferences(PREF_TAG_CONFERENCE, 0);
+        mRoomName           = settings.getString(PREF_ROOMNAME, "");
+        mUsername           = settings.getString(PREF_USERNAME, "");
+        mCapturerResolution = settings.getString(PREF_CAPTURERRES, "Medium");
+        mCapturerFps        = settings.getString(PREF_CAPTURERFPS, "30");
+        mH264Support        = settings.getBoolean(PREF_H264, true);
+        /* restore room name/user name UI */
+        ((EditText)findViewById(R.id.input_room_name)).setText(mRoomName);
+        ((EditText)findViewById(R.id.input_username)).setText(mUsername);
+        /* restore capturer resolution selection */
+        String[] resolution_lst = getResources().getStringArray(R.array.pub_capturer_resolution);
+        ((Spinner)findViewById(R.id.combo_capturer_resolution)).setSelection(
+                Arrays.asList(resolution_lst).indexOf(mCapturerResolution)
+        );
+        /* restore capturer frame reate selection */
+        String[] framerate_lst = getResources().getStringArray(R.array.pub_capturer_fps);
+        ((Spinner)findViewById(R.id.combo_capturer_fps)).setSelection(
+                Arrays.asList(framerate_lst).indexOf(mCapturerFps)
+        );
+        /* restore H264 support flag */
+        ((Switch)findViewById(R.id.h264Support)).setChecked(mH264Support);
+        /* Debug Settings */
+        settings = this.getSharedPreferences(PREF_TAG_DEBUG, 0);
+        OpenTokConfig.setJNILogs(settings.getBoolean(PREF_JNI_LOG, false));
+        OpenTokConfig.setOTKitLogs(settings.getBoolean(PREF_OTK_LOG, false));
+        OpenTokConfig.setOTKitLogs(settings.getBoolean(PREF_RTC_LOG, false));
+        OpenTokConfig.setUseMediaCodecFactories(settings.getBoolean(PREF_MEDIACODEC, false));
```

Comment:
You missed the following so H264 support is not working. Please add it and verify the app publish H264.
```
OpenTokConfig.setPreferH264Codec(mH264Support);
```